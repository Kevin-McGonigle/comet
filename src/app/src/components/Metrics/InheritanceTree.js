import React from 'react';
import { TreeGraphContainer } from '../general/TreeGraph/TreeGraphContainer';
import ClassDiagram from '../general/ClassDiagram/ClassDiagram';
import styles from './InheritanceTree.css';
import classnames from 'classnames';

const cx = args => classnames(styles, args)

const InheritanceTree = props => {
    return (
       <div className={cx('inheritanceTreeContainer')}>
           <TreeGraphContainer />
           <ClassDiagram />
       </div>
    )
};

export default InheritanceTree;
