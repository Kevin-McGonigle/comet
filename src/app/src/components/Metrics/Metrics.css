.metricContainer {
    display: flex;
    flex-direction: row;
    background: #333;
}

.metricGridContainer{
    margin: 24px;
}

.metricGrid {
    max-width:1000px;
    display: flex;

}
.logo {
    
}