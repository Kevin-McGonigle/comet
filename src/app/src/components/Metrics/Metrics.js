import React, { useState } from 'react';
import styles from './Metrics.css';
import classnames from 'classnames';
import Toolbar from '../general/Toolbar/Toolbar';
import { FileDirectoryContainer } from '../general/FileDirectory/FileDirectoryContainer';
import AbstractSyntaxTree from '../general/Charts/AbstractSyntaxTree';
import { classDiagramData } from '../general/Charts/configs';
import ForceDirectedGraph from '../general/Charts/ForceDirectedGraph';
import TreeMapContainer from '../general/TreeMap/TreeMapContainer';

const cx = args => classnames(styles, args)

const tabMapping = {
    "TreeMap": "TreeMap",
    "ClassDiagram": "Class Diagram",
    "InheritanceTree": "Inheritance Tree",
    "AbstractSyntaxTree": "Abstract Syntax Tree",
    "ControlFlowDiagram": "Control Flow Diagram",
    "DependencyGraph": "Dependency Graph",
}

const Metrics = props => {
    const {fileDirectory, selected, metrics } = props;
    const selectedMetrics = Object.values(metrics).filter(value => value.fileName === selected)[0]

    const [tabState, setTabState] = useState({
        selectedIndex: 0,
        tabs: Object.values(tabMapping),
        tabContent: [
            <TreeMapContainer metrics={selectedMetrics.metrics} />,
            <ForceDirectedGraph
                title="Class Diagram"
                data={classDiagramData}
                graphType="classDiagram"
            />,
            <ForceDirectedGraph
                title="Inheritance Tree"
                data={selectedMetrics.structures.inheritanceTree}
                graphType="dynamic"
            />,
            <AbstractSyntaxTree data={selectedMetrics.structures.abstractSyntaxTree} />, 
            <ForceDirectedGraph 
                title="Control Flow Graph"
                data={selectedMetrics.structures.controlFlowGraph} 
                graphType="dynamic"         
            />,
            <ForceDirectedGraph 
                title="Dependency Graph"
                data={selectedMetrics.structures.dependencyGraph} 
                graphType="dependencyGraph"         
            />
        ]
    });

    return (
        <>
            { fileDirectory && <FileDirectoryContainer /> }
            <div className={cx("metricContainer")}>
                <Toolbar tabMapping={tabMapping} tabState={tabState} setTabState={setTabState} />
                <div className={cx("metricTabContainer")}>
                    { tabState.tabContent[tabState.selectedIndex] }
                </div>
            </div>
        </>
    )
};

export default Metrics;