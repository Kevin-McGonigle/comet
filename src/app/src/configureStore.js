import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import createRootReducer from './store/reducers';

export default function configureStore(initialState = {}) {
  let composeEnhancers = compose;
  let reduxSagaMonitorOptions = {};

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    if (window.__SAGA_MONITOR_EXTENSION__)
      reduxSagaMonitorOptions = {
        sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
      };
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  const middlewares = [sagaMiddleware];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createRootReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  if (module.hot) {
    module.hot.accept('./store/reducers', () => {
      store.replaceReducer(createRootReducer(store.injectedReducers));
    });
  }

  return store;
}
