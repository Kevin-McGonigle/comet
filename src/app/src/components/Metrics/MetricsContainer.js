import {connect} from 'react-redux';
import Metrics from './Metrics';

const mapStateToProps = state => ({
    fileDirectory: true,
    selected: state.fileData.selected,
    metrics: state.metrics,
})

export const MetricsContainer = connect(mapStateToProps, null)(Metrics)



