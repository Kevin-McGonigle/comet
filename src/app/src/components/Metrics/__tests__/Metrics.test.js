import React from 'react';
import {Provider} from 'react-redux'
import Metrics from '../Metrics';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store'
import "../../../setupTests"


const state = {
    fileData: {
        selected: 'file',
        files: [{"file": {"name": "file"}}]
    },
    metrics: [
        {
            "fileName": "file",
            "structures": {
                "controlFlowGraph": null,
                "classDiagram": null,
                "inheritanceTree": null,
                "abstractSyntaxTree": null,
            },
            "metrics": {
                "dependencyGraph": null,
                "afferentCoupling": null,
                "efferentCoupling": null,
                "logicalLinesOfCode": null,
                "cyclomaticComplexity": null,
                "maximumInheritanceDepth": null,
                "maximumNestingDepth": null,
            }
        }
    ]
}
const mockStore = configureStore(state);

test('Should render Metrics successfully', () => {
    const component = shallow(
        <Provider store={mockStore()}>
            <Metrics fileDirectory={false} selected={state.fileData.selected} metrics={state.metrics}/>
        </Provider>
    )
    expect(component.html()).toMatchSnapshot()
})
