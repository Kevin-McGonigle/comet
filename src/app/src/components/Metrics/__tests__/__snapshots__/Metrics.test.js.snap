// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Metrics succesfully 1`] = `
"<div class=\\"metricContainer\\"><div class=\\"tabMenu\\"><div class=\\"css-nil ðŸ“¦mt_12px ðŸ“¦mb_12px ðŸ“¦ml_300px ðŸ“¦box-szg_border-box\\"><div role=\\"tablist\\" class=\\"ðŸ“¦box-szg_border-box\\"><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"TreeMap\\" aria-controls=\\"panel-TreeMap\\" aria-selected=\\"true\\" role=\\"tab\\">TreeMap</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Class Diagram\\" aria-controls=\\"panel-Class Diagram\\" aria-selected=\\"false\\" role=\\"tab\\">Class Diagram</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Inheritance Tree\\" aria-controls=\\"panel-Inheritance Tree\\" aria-selected=\\"false\\" role=\\"tab\\">Inheritance Tree</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Abstract Syntax Tree\\" aria-controls=\\"panel-Abstract Syntax Tree\\" aria-selected=\\"false\\" role=\\"tab\\">Abstract Syntax Tree</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Control Flow Diagram\\" aria-controls=\\"panel-Control Flow Diagram\\" aria-selected=\\"false\\" role=\\"tab\\">Control Flow Diagram</span><span class=\\"css-11bc0q4 ðŸ“¦color_white ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_28px ðŸ“¦ltr-spc_0 ðŸ“¦h_28px ðŸ“¦dspl_inline-flex ðŸ“¦pl_8px ðŸ“¦pr_8px ðŸ“¦ml_4px ðŸ“¦mr_4px ðŸ“¦bblr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦btrr_3px ðŸ“¦algn-itms_center ðŸ“¦just-cnt_center ðŸ“¦txt-deco_none ðŸ“¦box-szg_border-box\\" tabindex=\\"0\\" id=\\"Dependency Graph\\" aria-controls=\\"panel-Dependency Graph\\" aria-selected=\\"false\\" role=\\"tab\\">Dependency Graph</span></div></div><div class=\\"menu\\"><button class=\\"css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_16px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦box-szg_border-box\\" role=\\"button\\" aria-haspopup=\\"true\\">Menu</button></div></div><div class=\\"metricTabContainer\\"><div class=\\"css-nil ðŸ“¦bs_kymczs ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-wrap_wrap ðŸ“¦flx-grow_1 ðŸ“¦mb_10 ðŸ“¦ml_10 ðŸ“¦mr_10 ðŸ“¦mt_10 ðŸ“¦box-szg_border-box\\"><button class=\\"css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_16px ðŸ“¦pl_16px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_inline-flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦flt_right ðŸ“¦box-szg_border-box\\" role=\\"button\\" aria-haspopup=\\"true\\">Select metric</button><div><div class=\\"sc-bdVaJa cYYVwO\\"><a id=\\"0\\">flare</a></div><svg class=\\"TreeMap.module_mainSvg_3UpOl\\" height=\\"450\\" width=\\"1000\\"><g transform=\\"translate(0,0)\\" class=\\"Node.module_node_2sWtP null\\" id=\\"0\\" style=\\"cursor:auto\\"><rect id=\\"rect-0\\" width=\\"1000\\" height=\\"450\\" fill=\\"rgb(247, 252, 245)\\"></rect><clipPath id=\\"clip-myTreeMap-0\\"><rect width=\\"983\\" height=\\"450\\"></rect></clipPath><a target=\\"_blank\\"><text clip-path=\\"url(#clip-myTreeMap-0)\\"><tspan font-size=\\"14\\" fill=\\"black\\" dx=\\"4\\" dy=\\"14\\">flareÂ (50 MB)</tspan></text></a><g><rect id=\\"rectNumberItems-flare\\" x=\\"986\\" y=\\"2\\" width=\\"12\\" height=\\"14\\" fill=\\"rgb(247, 252, 245)\\" fill-opacity=\\"0.9\\" stroke=\\"black\\"></rect><text font-size=\\"14\\" fill=\\"black\\" x=\\"988\\" y=\\"14\\">5</text></g><title>flare
50 MB
5/6</title></g><g transform=\\"translate(3,19)\\" class=\\"Node.module_node_2sWtP null\\" id=\\"1\\" style=\\"cursor:pointer\\"><rect id=\\"rect-1\\" width=\\"994\\" height=\\"428\\" fill=\\"rgb(34, 139, 69)\\"></rect><clipPath id=\\"clip-myTreeMap-1\\"><rect width=\\"977\\" height=\\"428\\"></rect></clipPath><a target=\\"_blank\\"><text clip-path=\\"url(#clip-myTreeMap-1)\\"><tspan font-size=\\"14\\" fill=\\"black\\" dx=\\"4\\" dy=\\"14\\">analyticsÂ (50 MB)</tspan></text></a><g><rect id=\\"rectNumberItems-analytics\\" x=\\"980\\" y=\\"2\\" width=\\"12\\" height=\\"14\\" fill=\\"rgb(34, 139, 69)\\" fill-opacity=\\"0.9\\" stroke=\\"black\\"></rect><text font-size=\\"14\\" fill=\\"black\\" x=\\"982\\" y=\\"14\\">4</text></g><title>analytics
50 MB
4/6</title></g></svg></div></div></div></div>"
`;
